### ISSUE TITLE
Not detecting rust files automatically

### ISSUE THREAD

**Issue by ishankhare07** [rust-lang/rust.vim#46]
need to set it with `:set syntax=rust`
just in case, [here](https://github.com/ishankhare07/dotfiles/blob/master/.vimrc) is my `vimrc`

**Comment by tndoan** [comment]
I have checked your ```vimrc``` file. I think the problem is that you have set 

```
filetype off
```

so ```rust.vim``` cannot detect the file type. Check out the [document](http://vimdoc.sourceforge.net/htmldoc/filetype.html) of ```filetype``` for more information.

**Comment by ishankhare07** [comment]
the `filetype off` line is used because vundle requires it [see point 3 of vundle readme](https://github.com/VundleVim/Vundle.vim/blob/master/README.md)
i've [switched it back on](https://github.com/ishankhare07/dotfiles/blob/master/.vimrc#L47) after loading vundle

**Comment by tndoan** [comment]
Actually, I think it is the problem of Vundle. Searching around and people also [report](https://github.com/wavded/vim-stylus/issues/26) this kind of error. To overcome this problem, you can add the following line at the end of your ```.vimrc``` file

```
autocmd BufNewFile,BufRead *.rs set filetype=rust
```

When you open a new file or existing file whose extension is ```rs```, ```vim``` will automatically load ```rust``` plugin. I have tested and it works.

**Comment by chris-morgan** [comment]
@tndoan that’s what the `ftdetect/rust.vim` is for. Putting it in your .vimrc is not necessary.

Use `:filetype` to see what the current value of “detection” is. `:filetype plugin indent on` *should* make it all work just fine (c.f. the `:filetype-overview` help).

**Comment by bitrauser** [comment]
I use Vundle and I have to report the same problem. Like tndoan said calling `filetype off` at the beginning is required by Vundle itself and turned back on later. 

**OSX**: On osx I noticed that rust files were detected but I had to explicitly call `syntax on` _after_ Vundles init. After changing this, everything worked as intended.

**Windows 10 64bit**: I installed and configured Vundle (NERDtree works, for example), but it won't detect rust files, and I remain unable to fix this issue.

**Comment by unaffiliated** [comment]
Bump. 

I had to go with @tndoan 's solution since it seems `ftdetect/rust.vim` isn't being recognized. GNU/Linux Debian 8 here using Vundle.

**Comment by markotikvic** [comment]
I use Pathogen and rust.vim was not being loaded until I added @tndoan solution.
Thanks.

**Comment by da-x** [comment]
I tried a Vundle vimrc config from scratch with latest rust.vim and the issue was not reproduced. I am guessing it was an old bug that fixed in either projects. Please re-open if you see evidence it's not the case. Thanks!

**Comment by steavot** [comment]
Hi,

I've hit the same symptom of rust file not being detected.  Using:
```
OS: Centos7
vim: 8.0.3
cargo 1.33.0 (f099fe94b 2019-02-12)

rust.vim HEAD:
commit 2234f0d3f8204b44f30ff6eb3b6f7345c23aef92
Author: Jan Alexander Steffens <jan.steffens@gmail.com>
Date:   Thu Mar 28 20:31:16 2019 +0100

Vundle.vim HEAD
commit 9a38216a1c0c597f978d73547d37681fc689c90d (HEAD -> master, origin/master, origin/HEAD)
Merge: fcc2042 d015ffa
Author: Ryan L McIntyre <8083459+ryanoasis@users.noreply.github.com>
Date:   Fri Feb 2 22:16:35 2018 -0500
```
and [my vimrc](https://github.com/steavot/dotfiles/blob/master/.vimrc)

Running `:set sytax=rust` fixes the problem.
Including `autocmd BufNewFile,BufRead *.rs set filetype=rust` in my `.vimrc` fixes the problem.

Strangely I *don't see this issue* with my setup at home on a Debian based distribution:
```
Same `.vimrc` 
same colorscheme.
Same git HEAD in ~/.vim/bundles/rust.vim/
OS: Lubuntu
vim: 8.0.1453
```
not sure what vundle HEAD I have at home but can check this evening.

This to me suggests that it's something exterior to `rust.vim` that's causing this.  
Could there be something different in the way vim is installed from an RPM then a DEB that causes this?
I noticed my CentOS setup was missing the file `/usr/share/vim/vim80/rust.vim`, and in general had fewer files in that directory than my Lubuntu machine.
I've copied over  `/usr/share/vim/vim80/rust.vim` from my Lubuntu to CentOS but that didn't fixed the problem.

Both workarounds above fix the issue, so this isn't a particularly pressing matter, but I'm interested in learning how to debug problems like these further.
